### PR TITLE
Use sizeof instead of Marshal.Sizeof

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by Therzok** [comment]
I've not migrated `public` methods to unmanaged constraint. The internal users don't use any kind of special marshal attributes.

Regardless, I could add tests.

**Comment by Therzok** [comment]
Any suggestion of where to put the tests?

**Comment by spouliot** [comment]
@Therzok by default we add them to `monotouch-test` which is executed under most possible configurations.

File: src/AudioUnit/AudioUnit.cs [hunk 1]
```diff
@@ -644,7 +644,7 @@ public unsafe AudioUnitParameterInfo[] GetParameterList (AudioUnitScopeType scop
 			}
 
 			var info = new AudioUnitParameterInfo [data.Length];
-			size = (uint) Marshal.SizeOf (typeof (AudioUnitParameterInfoNative));
+			size = (uint)sizeof (AudioUnitParameterInfoNative);
```
Comment (Therzok) [inline src/AudioUnit/AudioUnit.cs]: This is internal, so I'm not sure how to test that.

**Comment by Therzok** [comment]
I've updated everything except for `AudioUnitParameterNative`, as that's internal

File: src/AudioUnit/AudioUnit.cs (hunk 1 above)
Comment (Therzok) [reply src/AudioUnit/AudioUnit.cs]: But the definition definitely should return the same values:
https://github.com/xamarin/xamarin-macios/blob/b8f9dccdcc09f43a44d591fd0eaf9ba954ea2752/src/AudioUnit/AudioUnit.cs#L235

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
failure is a known issue (HE0038) https://github.com/xamarin/maccore/issues/581

**Pull request merged**